src/ffi/cpython/object.rs

```diff
@@ -0,0 +1,483 @@
+                    tp_as_mapping: ptr::null_mut(),
+                    tp_hash: None,
+                    tp_call: None,
+                    tp_str: None,
+                    tp_getattro: None,
+                    tp_setattro: None,
+                    tp_as_buffer: ptr::null_mut(),
+                    tp_flags: object::Py_TPFLAGS_DEFAULT,
+                    tp_doc: ptr::null(),
+                    tp_traverse: None,
+                    tp_clear: None,
+                    tp_richcompare: None,
+                    tp_weaklistoffset: 0,
+                    tp_iter: None,
+                    tp_iternext: None,
+                    tp_methods: ptr::null_mut(),
+                    tp_members: ptr::null_mut(),
+                    tp_getset: ptr::null_mut(),
+                    tp_base: ptr::null_mut(),
+                    tp_dict: ptr::null_mut(),
+                    tp_descr_get: None,
+                    tp_descr_set: None,
+                    tp_dictoffset: 0,
+                    tp_init: None,
+                    tp_alloc: None,
+                    tp_new: None,
+                    tp_free: None,
+                    tp_is_gc: None,
+                    tp_bases: ptr::null_mut(),
+                    tp_mro: ptr::null_mut(),
+                    tp_cache: ptr::null_mut(),
+                    tp_subclasses: ptr::null_mut(),
+                    tp_weaklist: ptr::null_mut(),
+                    tp_del: None,
+                    tp_version_tag: 0,
+                    tp_finalize: None,
+                    #[cfg(Py_3_8)]
+                    tp_vectorcall: None,
+                    $($tail)*
+                }
+            }
+        }
+    }
+
+    #[cfg(PyPy)]
+    macro_rules! type_object_init {
+        ($($tail:tt)*) => {
+            _type_object_init!({
+                    ob_refcnt: 1,
+                    ob_pypy_link: 0,
+                    ob_type: ptr::null_mut(),
+                    ob_size: 0,
+                },
+                tp_pypy_flags: 0,
+                $($tail)*
+            )
+        }
+    }
+
+    #[cfg(not(PyPy))]
+    macro_rules! type_object_init {
+        ($($tail:tt)*) => {
+            _type_object_init!({
+                ob_base: object::PyVarObject {
+                    ob_base: object::PyObject_HEAD_INIT,
+                    ob_size: 0
+                },},
+                $($tail)*
+            )
+        }
+    }
+
+    #[cfg(py_sys_config = "COUNT_ALLOCS")]
+    #[deprecated(note = "not present in Python headers; to be removed")]
+    pub const PyTypeObject_INIT: PyTypeObject = type_object_init! {
+        tp_allocs: 0,
+        tp_frees: 0,
+        tp_maxalloc: 0,
+        tp_prev: ptr::null_mut(),
+        tp_next: ptr::null_mut(),
+    };
+
+    #[cfg(not(py_sys_config = "COUNT_ALLOCS"))]
+    #[deprecated(note = "not present in Python headers; to be removed")]
+    pub const PyTypeObject_INIT: PyTypeObject = type_object_init!();
```

Comment:
TBH I think this falls under the same bracket of the other `_INIT` constants which aren't actually defined upstream; we can just throw it away for the 0.14 release. Nobody should be using it.

Looks like that would allow quite a lot of macro code above to be deleted too!

Reply:
I thought so, but didn't want to presume. Updated.

Shall we also remove `PyObject_Check`, `PySuper_Check`, and `FreeFunc`? They're not in the Python API either, and I struggle to imagine why, for example, a downstream user would need need a function to verify that a PyObject is a Python object (always returns 1).

Reply:
Agreed that we can remove those three too!